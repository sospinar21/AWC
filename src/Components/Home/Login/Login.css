.signin {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 1px;
  margin: 0 0 10px;
  padding: 30px 35px; }

.policy {
  margin-top: 30px; }

.website {
  box-shadow: -1px 1px 13px lavenderblush;
  background-image: url("http://www.marinecitydancecompany.com//Cut%20out%20images/RED%20Jump%20cropped.png");
  background-size: cover;
  position: absolute;
  height: 800px;
  width: 45vw;
  margin: 4% 240px 0;
  background-repeat: no-repeat; }
