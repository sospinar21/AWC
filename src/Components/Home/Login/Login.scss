.forms {
  width: 30%;
  margin-left: 55%;
  padding: 10px;
  box-sizing: border-box;
  text-align: center;
  position: relative;

  button, input{
    display:block;
    border: .5px solid lightgrey;
    -webkit-box-flex: 1;
    -webkit-flex: 1 0 0;
    -ms-flex: 1 0 0;
    flex: 1 0 0px;
    margin-top: 20px;
    outline: 0;
    overflow: hidden;
    padding: 9px 0 7px 8px;
    text-overflow: ellipsis;
    background-color: #fafafa;
    width: 100%;
    box-sizing: border-box;
    height: 60px;
    font-size: 15px;
  }

  h1{
    color: #232b37;
    font-size: 50px;
    font-family: 'Kalam', cursive;
    text-shadow: 1px 1px 1px #100b0b;
  }

  button {
    background-color: #b62d4d;
    color: white;
    height: 50px;
    
  }
}

.signup-box {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 1px;
  margin: 0 0 10px;
  padding: 30px 35px;
}

.form {

}

.email

.password {}

.city {

}

.signup {

}

.email-si {

}

.signin{
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 1px;
  margin: 0 0 10px;
  padding: 30px 35px;
} 

.password-si{}

.si-su {
  height: 1000px;
  overflow-y: hidden;
  background-color: #d3d3d347;
}

.policy {
  margin-top: 30px
}

.website {
  box-shadow: -1px 1px 13px lavenderblush;
  background-image: url('http://www.marinecitydancecompany.com//Cut%20out%20images/RED%20Jump%20cropped.png');
  background-size: cover;
  position: absolute;
  height: 900px;
  width: 45vw;
  margin: 4% 240px 0;
  background-repeat: no-repeat;
}