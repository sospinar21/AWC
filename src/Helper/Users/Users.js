import { Config, CognitoUserPool, CognitoUserAttribute, CognitoUser, AuthenticationDetails, CognitoIdentityCredentials } from 'amazon-cognito-identity-js';
import { Lambda } from 'aws-sdk';

var poolData = { UserPoolId : 'us-west-2_t4LvOKjcE',
  ClientId: '47bsfajnf2rmvpt02q8qjvm29u'
};
var userPool = new CognitoUserPool(poolData);

export const checkUser = () => {
  var cognitoUser = userPool.getCurrentUser();
  return cognitoUser;
}

export const signUp = (user) => {
  var attributeList = [];

  var dataEmail = {
    Name : 'email',
    Value : user.email
  };
  var city = {
    Name : 'custom:city',
    Value : user.city
  };
  var attributeEmail = new CognitoUserAttribute(dataEmail);
  var attributeCity = new CognitoUserAttribute(city);

  attributeList.push(attributeEmail);
  attributeList.push(attributeCity);

  userPool.signUp(user.email, user.password, attributeList, null, function(err, result) {
    if (err) {
      const error = document.querySelector('.created');
      error.innerText = 'invalid email or password: min 6 characters';
      return;
    }
    var cognitoUser = result.user;
    const created = document.querySelector('.created');
    created.innerText = 'Welcome! your new user name is ' + cognitoUser.getUsername();
    window.location.href = "http://awc.dance/";
  });
};

export const logIn = (user) => {
  var userData = {Username: user.email, Pool: userPool};
  var cognitoUser = new CognitoUser(userData);

  var authenticationData = {Username: user.email, Password: user.password};
  var authenticationDetails = new AuthenticationDetails(authenticationData);

  cognitoUser.authenticateUser(authenticationDetails, {
    onSuccess: function (result) {
      const success = document.querySelector('.success');
      success.innerText = 'Welcome back! ' + cognitoUser.getUsername();
      window.location.href = "http://awc.dance/";
      return result; 
    },

    onFailure: function (err) {
      const incorrect = document.querySelector('.success');
      incorrect.innerText = 'Check your Password or Email';
      return err; 
    }

  });
};

export const forgotPassword = () => {
  CognitoUser.forgotPassword({
    onSuccess: function (result) {
      console.log('call result: ' + result);
    },
    onFailure: function(err) {
      alert(err);
    },
    inputVerificationCode() {
      var verificationCode = prompt('Please input verification code ', '');
      var newPassword = prompt('Enter new password ', '');
      CognitoUser.confirmPassword(verificationCode, newPassword, this);
    }
  });
};

export const userSignout = () => {
  var cognitoUser = userPool.getCurrentUser();
  console.log(cognitoUser)
  cognitoUser.signOut();
  console.log(cognitoUser)
  return cognitoUser;
}

