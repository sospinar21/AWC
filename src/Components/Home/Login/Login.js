import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './Login.css';
import {  logIn } from '../../../Helper/Users/Users';
import { NavBar } from '../NavBar/NavBar';
import { addUser } from '../../../Actions/actions';
import {CognitoUserPool} from 'amazon-cognito-identity-js';
import { Redirect } from 'react-router';
import ApiCalls from '../../../Helper/ApiCalls/ApiCalls';



export class Login extends Component {
  constructor() {
    super();
    this.state = {
      login: false,
      email: '',
      password: ''
    };
  }

  handleInputChange = (e) => {
    const {name, value} = e.target;
    this.setState({[name]: value});
  }

  userSignIn = (e) => {
    e.preventDefault();
    logIn(this.state);
    this.getToken();
    this.setState({email:'', password:''});
  }

  validateEmail = () => {
    const {email, password} = this.state;
    return (
      email.length === 0,
      password.length === 0
    );
  }

  getToken = () => {
    var poolData = { UserPoolId : 'us-west-2_t4LvOKjcE',
      ClientId: '47bsfajnf2rmvpt02q8qjvm29u'
    };

    var userPool = new CognitoUserPool(poolData);
    var cognitoUser = userPool.getCurrentUser();

    if (cognitoUser != null) {
      cognitoUser.getSession((err, session) =>  {
        if (err) {
          alert(err);
          return;
        }
        var token = session.getIdToken().getJwtToken();
        this.giveAccess(cognitoUser);
      });
    }
  }

  giveAccess = async (user) => {
    this.props.addUser(user);
    this.setState({login:true});
  }

  render () {

    this.state.login === true && <Redirect to='/profile' />;
    
    return (
      <div className='si-su'>
        <div className='signin'>
          <h3>LOGIN</h3>
          <div className="logo">
            <div className="logocir">AWC</div>
          </div>
          <h4 className='success'></h4>
          <h4 className='incorrect'></h4>
          <form className='login-form' autoComplete='on'>
            <input 
              name='email'
              type='email'
              onChange={this.handleInputChange}
              value={this.state.email} 
              autoComplete='on'
              placeholder='Email'             
              className='email-si'/>
            <input 
              name='password'
              onChange={this.handleInputChange}
              value={this.state.password} 
              autoComplete='on'
              type='password' 
              placeholder='Password'
              className='password-si'/>
            <button
              disabled={this.validateEmail()}  
              onClick={(e) => this.userSignIn(e)}              
              className='login'>Log In</button>              
          </form>
        </div>
      </div>

    );
  }
}

export const mapStateToProps = (state) => {
  return ({
    suggestedEvents: state.suggestedEvents,
    user: state.user
  });
};

export const mapDispatchToProps = dispatch => ({
  addUser: (user) => dispatch(addUser(user))
});

Login.propTypes = {
  suggestedEvents: PropTypes.array,
  addUser: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);