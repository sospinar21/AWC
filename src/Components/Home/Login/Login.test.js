import {Login, mapStateToProps, mapDispatchToProps} from '../Login/Login';
import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import {logIn} from '../../../Helper/Users/Users';

describe('login', () => {

  let login;
  let mockProps;
  let event;

  beforeEach(() => {

    mockProps = {
      suggestedEvents: [{name: 'party'}],
      user: {namename:'Steph'},
      addUser: jest.fn()
    };

    login = shallow(<Login {...mockProps}/>, { disableLifecycleMethods: true });

    event = {
      target: {
        email: "user@user.com",
        password: "hola"
      },
      preventDefault: () => {}
    };
  });

  it('matches snapshot', () => {
    expect(login).toMatchSnapshot();
  });


  // describe('userSignIn', () => {

  //   it('sets the state with the new users email and psw', () => {

  //     login.state().email = '';

  //     login.instance().userSignIn(event);

  //     expect(login.state().email).toEqual('a@a.me');
  //   });
  // });

  describe('validateEmail', () => {

    it('returns true if the length of email or password is 0', () => {

      login.state().email = '';
      login.state().password = '';

      expect(login.instance().validateEmail()).toEqual(true);
    });

    it('returns false if the length of email or password is 0', () => {

      login.state().email = 'me@me';
      login.state().password = 'hola';

      expect(login.instance().validateEmail()).toEqual(false);
    });
  });

  // describe('giveAccess', () => {

  //   it('calls giveAccess with the correct params', () => {
  
  //     login.instance().getToken();
  //     login.instance().giveAccess = jest.fn()
  
  //     expect(login.instance().giveAccess).toHaveBeenCalled()
  //   });
  // });
});