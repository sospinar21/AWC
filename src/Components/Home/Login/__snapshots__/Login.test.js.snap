// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    addUser={[Function]}
    suggestedEvents={
        Array [
            Object {
              "name": "party",
            },
          ]
    }
    user={
        Object {
            "namename": "Steph",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="signin"
>
        <h3>
                LOGIN
        </h3>
        <h1>
                AWC 
        </h1>
        <h4
                className="success"
        />
        <h4
                className="incorrect"
        />
        <form
                autoComplete="on"
                className="login-form"
        >
                <input
                        autoComplete="on"
                        className="email-si"
                        name="email"
                        onChange={[Function]}
                        placeholder="Email"
                        type="email"
                        value=""
                />
                <input
                        autoComplete="on"
                        className="password-si"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
                />
                <button
                        className="login"
                        disabled={true}
                        onClick={[Function]}
                >
                        Log In
                </button>
        </form>
</div>,
      "className": "si-su",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            LOGIN
</h3>,
          <h1>
            AWC 
</h1>,
          <h4
            className="success"
/>,
          <h4
            className="incorrect"
/>,
          <form
            autoComplete="on"
            className="login-form"
>
            <input
                        autoComplete="on"
                        className="email-si"
                        name="email"
                        onChange={[Function]}
                        placeholder="Email"
                        type="email"
                        value=""
            />
            <input
                        autoComplete="on"
                        className="password-si"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
            />
            <button
                        className="login"
                        disabled={true}
                        onClick={[Function]}
            >
                        Log In
            </button>
</form>,
        ],
        "className": "signin",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "LOGIN",
          },
          "ref": null,
          "rendered": "LOGIN",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "AWC ",
          },
          "ref": null,
          "rendered": "AWC ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "success",
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "incorrect",
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "on",
            "children": Array [
              <input
                autoComplete="on"
                className="email-si"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="email"
                value=""
/>,
              <input
                autoComplete="on"
                className="password-si"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
/>,
              <button
                className="login"
                disabled={true}
                onClick={[Function]}
>
                Log In
</button>,
            ],
            "className": "login-form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoComplete": "on",
                "className": "email-si",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoComplete": "on",
                "className": "password-si",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log In",
                "className": "login",
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Log In",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="signin"
>
          <h3>
                    LOGIN
          </h3>
          <h1>
                    AWC 
          </h1>
          <h4
                    className="success"
          />
          <h4
                    className="incorrect"
          />
          <form
                    autoComplete="on"
                    className="login-form"
          >
                    <input
                              autoComplete="on"
                              className="email-si"
                              name="email"
                              onChange={[Function]}
                              placeholder="Email"
                              type="email"
                              value=""
                    />
                    <input
                              autoComplete="on"
                              className="password-si"
                              name="password"
                              onChange={[Function]}
                              placeholder="Password"
                              type="password"
                              value=""
                    />
                    <button
                              className="login"
                              disabled={true}
                              onClick={[Function]}
                    >
                              Log In
                    </button>
          </form>
</div>,
        "className": "si-su",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              LOGIN
</h3>,
            <h1>
              AWC 
</h1>,
            <h4
              className="success"
/>,
            <h4
              className="incorrect"
/>,
            <form
              autoComplete="on"
              className="login-form"
>
              <input
                            autoComplete="on"
                            className="email-si"
                            name="email"
                            onChange={[Function]}
                            placeholder="Email"
                            type="email"
                            value=""
              />
              <input
                            autoComplete="on"
                            className="password-si"
                            name="password"
                            onChange={[Function]}
                            placeholder="Password"
                            type="password"
                            value=""
              />
              <button
                            className="login"
                            disabled={true}
                            onClick={[Function]}
              >
                            Log In
              </button>
</form>,
          ],
          "className": "signin",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "LOGIN",
            },
            "ref": null,
            "rendered": "LOGIN",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "AWC ",
            },
            "ref": null,
            "rendered": "AWC ",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "success",
            },
            "ref": null,
            "rendered": null,
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "incorrect",
            },
            "ref": null,
            "rendered": null,
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoComplete": "on",
              "children": Array [
                <input
                  autoComplete="on"
                  className="email-si"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  type="email"
                  value=""
/>,
                <input
                  autoComplete="on"
                  className="password-si"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
/>,
                <button
                  className="login"
                  disabled={true}
                  onClick={[Function]}
>
                  Log In
</button>,
              ],
              "className": "login-form",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "autoComplete": "on",
                  "className": "email-si",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "Email",
                  "type": "email",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "autoComplete": "on",
                  "className": "password-si",
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "Password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Log In",
                  "className": "login",
                  "disabled": true,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Log In",
                "type": "button",
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
